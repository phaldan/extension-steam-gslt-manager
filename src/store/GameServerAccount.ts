import { action, observable } from 'mobx';

export default class GameServerAccount {
  public steamid: string;
  public appid: number;

  @observable
  public expired: boolean = false;

  @observable
  public lastLogon?: Date;

  @observable
  public memo: string;

  @observable
  public token: string;

  @observable
  public locked: boolean = false;

  constructor({ steamid, appid, token, memo, lastLogon, expired }) {
    this.steamid = steamid;
    this.appid = appid;
    this.token = token;
    this.memo = memo;
    this.lastLogon = lastLogon ? new Date(lastLogon) : undefined;
    this.expired = expired;
  }

  @action
  public toggleLock(locked: boolean) {
    this.locked = locked;
  }

  @action
  public updateFromJson({ appid, token, memo, lastLogon, expired }) {
    this.appid = appid;
    this.token = token;
    this.memo = memo;
    this.lastLogon = lastLogon;
    this.expired = expired;
  }
}
