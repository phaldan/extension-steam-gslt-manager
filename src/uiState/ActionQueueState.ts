import { action, computed, observable } from 'mobx';
import ActionQueue from './ActionQueue';

export default class ActionQueueState {
  private removeDelay = 3000;

  @observable
  private queues: ActionQueue[] = [];

  @action
  public create(description: string, data: string[]): ActionQueue {
    const entry = new ActionQueue(this, this.removeDelay, description, data);
    this.queues.push(entry);
    return entry;
  }

  @computed
  public get running(): ActionQueue[] {
    return this.queues;
  }

  @action
  public async remove(id: string) {
    this.queues = this.queues.filter((q) => q.id !== id);
  }

  public setRemoveDelay(removeDelay: number): void {
    this.removeDelay = removeDelay;
  }
}
