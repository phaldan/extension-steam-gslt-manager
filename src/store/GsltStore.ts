import GameServerAccount from './GameServerAccount';

export default interface GsltStore {

  /**
   * Return current list of GameServerAccount.
   */
  readonly tokenAccounts: GameServerAccount[];

  readonly isLoggedIn: boolean;

  readonly isInitialized: boolean;

  /**
   * Load token accounts from steam.
   */
  loadAccounts(): Promise<void>;

  /**
   * Remove GSLT account.
   * @param account GameServerAccount instance
   */
  removeAccount(account: GameServerAccount): Promise<GameServerAccount>;

  /**
   * Remove GSLT accounts.
   * @param accounts List of GameServerAccount instances
   */
  removeAccounts(accounts: GameServerAccount[]): Array<Promise<GameServerAccount>>;

  /**
   * Create a new token for GSLT account.
   * @param account GameServerAccount instance
   */
  regenerateToken(account: GameServerAccount): Promise<GameServerAccount>;

  /**
   * Create a new token for GSLT accounts.
   * @param accounts List of GameServerAccount instances.
   */
  regenerateTokens(accounts: GameServerAccount[]): Array<Promise<GameServerAccount>>;

  /**
   * Change memo of GSLT account.
   * @param account GameServerAccount instance
   * @param memo Description for a GSLT
   */
  updateMemo(account: GameServerAccount, memo: string): Promise<GameServerAccount>;

  /**
   * Create multiple GSLT accounts.
   * @param amount Number of GSLT to create
   * @param appid Steam AppID
   * @param memo Description for a GSLT
   */
  createAccounts(amount: number, appid: string, memo: string): Array<Promise<void>>;
}
