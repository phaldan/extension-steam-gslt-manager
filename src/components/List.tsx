import { observer } from 'mobx-react';
import React from 'react';
import { Button, Checkbox, ControlLabel, FormGroup, Glyphicon } from 'react-bootstrap';
import GameServerAccount from '../store/GameServerAccount';
import GsltStore from '../store/GsltStore';
import ActionQueueState from '../uiState/ActionQueueState';
import SelectList from '../uiState/SelectList';
import Create from './Create';
import Entry from './Entry';
import Export from './Export';

interface ListProps {
  store: GsltStore;
  actions: ActionQueueState;
  items: GameServerAccount[];
  enableRegenerate?: boolean;
}

@observer
export default class List extends React.Component<ListProps> {
  public static defaultProps: Partial<ListProps> = {
    enableRegenerate: false,
  };

  private selections: SelectList;

  constructor(props) {
    super(props);
    this.selections = new SelectList();
  }

  public render() {
    const { store, items, enableRegenerate } = this.props;
    const { checked } = this.selections;
    return (
      <div>
        <div className="form-inline list-form-actions">
          <Button
            onClick={this.handleButtonSelect}
            className="js-select-button"
          >
            <Checkbox
              checked={checked}
              className="js-select-checkbox"
            />
          </Button>

          <FormGroup>
            <ControlLabel>
              Actions:
            </ControlLabel>
            <Button
              bsStyle="danger"
              title="Remove"
              onClick={this.handleRemove}
              className="js-remove"
            >
              <Glyphicon glyph="trash" /> Remove
            </Button>
            <Button
              bsStyle="success"
              title="Regenerate Token"
              onClick={this.handleRegenerate}
              disabled={!enableRegenerate}
              className="js-regenerate"
            >
              <Glyphicon glyph="refresh" /> Regenerate Token
            </Button>
            <Button
              bsStyle="primary"
              title="CSV"
              href={this.csv}
              download="gslt.csv"
              className="js-csv"
            >
              <Glyphicon glyph="download" /> CSV
            </Button>
            <Export accounts={this.selectedAccounts} />
          </FormGroup>
        </div>

        <table className="table table-striped table-hover table-responsive table-condensed">
          <thead>
            <tr>
              <th/>
              <th className="text-center">Token</th>
              <th className="text-center">Game</th>
              <th className="text-center">Last logon</th>
              <th className="text-center">Memo</th>
              <th/>
            </tr>
          </thead>
          <tbody>
            {items.length > 0 ? this.createEntries : this.createMessage}
          </tbody>
        </table>
      </div>
    );
  }

  private isSelected(account: GameServerAccount) {
    const selection = this.selections.all.find((s) => s.key === account.steamid);
    return (selection === undefined) ? false : selection.selected;
  }

  private handleEntryToggle = (checked: boolean, account: GameServerAccount) => {
    this.selections.set(account.steamid, checked);

    if (this.selections.checked && !checked) {
      this.selections.toggleChecked(false);
    }
  }

  private handleButtonSelect = () => {
    const { checked } = this.selections;
    this.handleCheckboxToggle(!checked);
  }

  private handleCheckboxToggle(checked: boolean) {
    const { selectAll } = this.selections;
    this.toggleSelectAll(selectAll === checked ? !checked : checked);
  }

  private toggleSelectAll(checked: boolean) {
    const { items } = this.props;

    this.selections.clearSelections();
    for (const account of items) {
      this.selections.add(account.steamid, checked);
    }

    this.selections.toggleChecked(checked);
    this.selections.toggleSelectAll(checked);
  }

  private get selectedAccounts() {
    const { items } = this.props;
    const steamids = this.selections.selected.map((s) => s.key);
    const selected = items.filter((a) => steamids.indexOf(a.steamid) >= 0);
    return selected;
  }

  private handleRemove = async () => {
    const { store } = this.props;
    const selected = this.selectedAccounts;
    if (selected.length === 0) {
      return;
    }
    selected.forEach((s) => s.toggleLock(true));
    this.resetSelectAll();

    const queue = this.createQueue(selected, `Deleting ${selected.length} GSLT:`);
    const promises = store.removeAccounts(selected);
    const actions = this.handleQueueProgress(promises, queue);
    return Promise.all(actions);
  }

  private createQueue(selected, description) {
    const { actions } = this.props;
    const steamids = selected.map((e) => e.steamid);
    return actions.create(description, steamids);
  }

  private handleQueueProgress(promises, queue) {
    return promises.map(async (action) => {
      const gslt = await action;
      queue.progress(gslt.steamid);
    });
  }

  private handleRegenerate = async () => {
    const { store, items } = this.props;
    if (!this.props.enableRegenerate) {
      return;
    }
    const selected = this.selectedAccounts;
    if (selected.length === 0) {
      return;
    }
    this.resetSelectAll();

    const queue = this.createQueue(selected, `Regenerate token for ${selected.length} GSLT:`);
    const promises = store.regenerateTokens(selected);
    const actions = this.handleQueueProgress(promises, queue);
    return Promise.all(actions);
  }

  private resetSelectAll() {
    const { items } = this.props;
    if (items.length === this.selections.selected.length) {
      this.toggleSelectAll(false);
    }
  }

  private get csv() {
    const lines = this.selectedAccounts.map((account) => this.createCsvLine(account));
    const content = lines.join('\n');
    const encoded = encodeURIComponent(this.createCsvHeadline() + content);
    return 'data:text/csv;charset=UTF-8,' + '\uFEFF' + encoded;
  }

  private createCsvHeadline() {
    return `"SteamID","Token","AppID","Last-Logon","Memo"\n`;
  }

  private createCsvLine(account: GameServerAccount) {
    const logon = (account.lastLogon ? `"${account.lastLogon.toISOString()}"` : '');
    return `"${account.steamid}","${account.token}",${account.appid},${logon},"${account.memo}"`;
  }

  private get createEntries() {
    const { store, items, enableRegenerate, actions } = this.props;
    return items.map((account) => (
      <Entry
        key={account.steamid}
        store={store}
        account={account}
        selected={this.isSelected(account)}
        onToggle={this.handleEntryToggle}
        enableRegenerate={enableRegenerate}
        actions={actions}
      />
    ));
  }

  private get createMessage() {
    const { store, actions } = this.props;
    return (
      <tr>
        <td colSpan={6}>
          <div className="panel panel-info" style={{margin: 0}}>
            <div className="panel-heading">
              <div className="page-header text-center">
                <h2>Nothing found!</h2>
                <Create store={store} text="Create new" actions={actions} />
              </div>
            </div>
          </div>
        </td>
      </tr>
    );
  }
}
