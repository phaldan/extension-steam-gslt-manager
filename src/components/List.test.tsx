import { mount, shallow, ShallowWrapper } from 'enzyme';
import * as React from 'react';
import GameServerAccount from '../store/GameServerAccount';
import GsltStore from '../store/GsltStore';
import ActionQueueState from '../uiState/ActionQueueState';
import delay from '../utils/delay';
import List from './List';

class GsltStoreStub implements GsltStore {
  public removeAccountsMock = jest.fn();
  public removeAccountsReturn: Promise<GameServerAccount>[];
  public regenerateTokensMock = jest.fn();
  public regenerateTokensReturn: Promise<GameServerAccount>[];

  loadAccounts(): Promise<void> {
    throw new Error("Method not implemented.");
  }
  get tokenAccounts(): GameServerAccount[] {
    throw new Error("Method not implemented.");
  }
  removeAccount(account: GameServerAccount): Promise<GameServerAccount> {
    throw new Error("Method not implemented.");
  }
  removeAccounts(accounts: GameServerAccount[]): Promise<GameServerAccount>[] {
    this.removeAccountsMock(accounts);
    return this.removeAccountsReturn;
  }
  regenerateToken(account: GameServerAccount): Promise<GameServerAccount> {
    throw new Error("Method not implemented.");
  }
  regenerateTokens(accounts: GameServerAccount[]): Promise<GameServerAccount>[] {
    this.regenerateTokensMock(accounts);
    return this.regenerateTokensReturn;
  }
  updateMemo(account: GameServerAccount, memo: string): Promise<GameServerAccount> {
    throw new Error("Method not implemented.");
  }
  createAccounts(amount: number, appid: string, memo: string): Promise<void>[] {
    throw new Error("Method not implemented.");
  }
  get isLoggedIn(): boolean {
    throw new Error("Method not implemented.");
  }
  get isInitialized(): boolean {
    throw new Error("Method not implemented.");
  }
}

describe('<List>', () => {
  let store: GsltStoreStub;
  let queue: ActionQueueState;

  function createComponent(items, regenerate) {
    const element = (
      <List store={store} actions={queue} items={items} enableRegenerate={regenerate} />
    );
    return shallow(element);
  }

  function expectExport(target, accounts) {
    expect(target.find('Export').props().accounts).toEqual(accounts);
  }

  function expectCreate(target) {
    expect(target.find('Entry').exists()).not.toBeTruthy();
    expect(target.find('Create').props().store).toBe(store);
    expect(target.find('Create').props().actions).toBe(queue);
  }

  function expectCsv(target, rows = '') {
    const href = 'data:text/csv;charset=UTF-8,\uFEFF%22SteamID%22%2C%22Token%22%2C%22AppID%22%2C%22Last-Logon%22%2C%22Memo%22%0A';
    expect(target.find('.js-csv').props().href).toEqual(href + rows);
  }

  function expectEmptyListOfItems(target, disabledRegenerate: boolean, checkedSelectAll: boolean) {
    expect(target.find('.js-select-checkbox').props().checked).toEqual(checkedSelectAll);
    expect(target.find('.js-regenerate').props().disabled).toEqual(disabledRegenerate);
    expectCsv(target);
    expect(target.find('Button')).toMatchSnapshot();
    expectExport(target, []);
    expectCreate(target);
    expect(store.removeAccountsMock).not.toBeCalled();
    expect(store.regenerateTokensMock).not.toBeCalled();
    expect(queue.running).toHaveLength(0);
  }

  function expectEntry(target, account, isSelected) {
    expect(target.find('Create').exists()).not.toBeTruthy();
    const entry = target.find('Entry');
    expect(entry.props().store).toBe(store);
    expect(entry.props().account).toBe(account);
    expect(entry.props().selected).toBe(isSelected);
    expect(entry.props().enableRegenerate).toBe(true);
    expect(entry.props().actions).toBe(queue);
    
  }

  function expectWithAccount(target, account, checkedSelectAll, csv, isSelected, isLocked, selectedAccounts) {
    expect(target.find('.js-select-checkbox').props().checked).toEqual(checkedSelectAll);
    expect(target.find('.js-regenerate').props().disabled).toEqual(false);
    expectCsv(target, csv);
    expect(target.find('Button')).toMatchSnapshot();

    expectExport(target, selectedAccounts);
    expectEntry(target, account, isSelected);

    expect(account.locked).toEqual(isLocked);
    expect(queue.running).toHaveLength(0);
  }

  beforeEach(() => {
    store = new GsltStoreStub();
    queue = new ActionQueueState();
    queue.setRemoveDelay(0);
  });

  describe('empty items', () => {
    test('regenerate enabled', () => {
      const target = createComponent([], true);
      target.find('.js-regenerate').simulate('click');

      expectEmptyListOfItems(target, false, false);
    });

    test('regenerate disabled', () => {
      const target = createComponent([], false);
      target.find('.js-regenerate').simulate('click');

      expectEmptyListOfItems(target, true, false);
    });

    test('click remove', () => {
      const target = createComponent([], false);
      target.find('.js-remove').simulate('click');

      expectEmptyListOfItems(target, true, false);
    });

    test('select all via button click', () => {
      const target = createComponent([], false);
      target.find('.js-select-button').simulate('click');

      expectEmptyListOfItems(target, true, true);
    });
  });

  describe('with a single account', () => {
    let account: GameServerAccount;
    let target: ShallowWrapper;

    beforeEach(() => {
      account = new GameServerAccount({
        appid: 740,
        expired: false,
        lastLogon: '1995-12-17T03:24:00',
        memo: 'CSGO',
        steamid: '212V16ECZ4HE',
        token: '7FJS3VY2273L',
      });
      target = createComponent([account], true);
    });

    test('select all', () => {
      target.find('.js-select-button').simulate('click');

      const csv = '%22212V16ECZ4HE%22%2C%227FJS3VY2273L%22%2C740%2C%221995-12-17T03%3A24%3A00.000Z%22%2C%22CSGO%22';
      expectWithAccount(target, account, true, csv, true, false, [account]);
      expect(store.removeAccountsMock).not.toBeCalled();
      expect(store.regenerateTokensMock).not.toBeCalled();
    });

    test('deselect by entry', () => {
      target.find('.js-select-button').simulate('click');
      target.find('Entry').simulate('toggle', false, account);

      expectWithAccount(target, account, false, '', false, false, []);
      expect(store.removeAccountsMock).not.toBeCalled();
      expect(store.regenerateTokensMock).not.toBeCalled();
    });

    test('remove all', async () => {
      store.removeAccountsReturn = [Promise.resolve(account)];
      target.find('.js-select-button').simulate('click');
      target.find('.js-remove').simulate('click');

      await delay(10); // Waiting for ActionQueueState to finish
      expectWithAccount(target.update(), account, false, '', false, true, []);
      expect(store.removeAccountsMock).toBeCalledWith([account]);
      expect(store.regenerateTokensMock).not.toBeCalled();
    });

    test('regenerate all', async () => {
      store.regenerateTokensReturn = [Promise.resolve(account)];
      target.find('.js-select-button').simulate('click');
      target.find('.js-regenerate').simulate('click');
      await delay(10); // Waiting for ActionQueueState to finish
      expectWithAccount(target.update(), account, false, '', false, false, []);
      expect(store.removeAccountsMock).not.toBeCalled();
      expect(store.regenerateTokensMock).toBeCalledWith([account]);
    });
  });

  test('with undefined lastLogon', () => {
    const account = new GameServerAccount({
      appid: 740,
      expired: false,
      lastLogon: undefined,
      memo: 'CSGO',
      steamid: '212V16ECZ4HE',
      token: '7FJS3VY2273L',
    });
    const target = createComponent([account], true);
    target.find('.js-select-button').simulate('click');

    const csv = '%22212V16ECZ4HE%22%2C%227FJS3VY2273L%22%2C740%2C%2C%22CSGO%22';
    expectWithAccount(target.update(), account, true, csv, true, false, [account]);
    expect(store.removeAccountsMock).not.toBeCalled();
    expect(store.regenerateTokensMock).not.toBeCalled();
  });
});
