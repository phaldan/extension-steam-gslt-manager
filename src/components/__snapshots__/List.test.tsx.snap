// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List> empty items click remove 1`] = `
Array [
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    className="js-select-button"
    disabled={false}
    onClick={[Function]}
  >
    <Checkbox
      bsClass="checkbox"
      checked={false}
      className="js-select-checkbox"
      disabled={false}
      inline={false}
      title=""
    />
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="danger"
    className="js-remove"
    disabled={false}
    onClick={[Function]}
    title="Remove"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="trash"
    />
     Remove
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="success"
    className="js-regenerate"
    disabled={true}
    onClick={[Function]}
    title="Regenerate Token"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="refresh"
    />
     Regenerate Token
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="primary"
    className="js-csv"
    disabled={false}
    download="gslt.csv"
    href="data:text/csv;charset=UTF-8,﻿%22SteamID%22%2C%22Token%22%2C%22AppID%22%2C%22Last-Logon%22%2C%22Memo%22%0A"
    title="CSV"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="download"
    />
     CSV
  </Button>,
]
`;

exports[`<List> empty items regenerate disabled 1`] = `
Array [
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    className="js-select-button"
    disabled={false}
    onClick={[Function]}
  >
    <Checkbox
      bsClass="checkbox"
      checked={false}
      className="js-select-checkbox"
      disabled={false}
      inline={false}
      title=""
    />
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="danger"
    className="js-remove"
    disabled={false}
    onClick={[Function]}
    title="Remove"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="trash"
    />
     Remove
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="success"
    className="js-regenerate"
    disabled={true}
    onClick={[Function]}
    title="Regenerate Token"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="refresh"
    />
     Regenerate Token
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="primary"
    className="js-csv"
    disabled={false}
    download="gslt.csv"
    href="data:text/csv;charset=UTF-8,﻿%22SteamID%22%2C%22Token%22%2C%22AppID%22%2C%22Last-Logon%22%2C%22Memo%22%0A"
    title="CSV"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="download"
    />
     CSV
  </Button>,
]
`;

exports[`<List> empty items regenerate enabled 1`] = `
Array [
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    className="js-select-button"
    disabled={false}
    onClick={[Function]}
  >
    <Checkbox
      bsClass="checkbox"
      checked={false}
      className="js-select-checkbox"
      disabled={false}
      inline={false}
      title=""
    />
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="danger"
    className="js-remove"
    disabled={false}
    onClick={[Function]}
    title="Remove"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="trash"
    />
     Remove
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="success"
    className="js-regenerate"
    disabled={false}
    onClick={[Function]}
    title="Regenerate Token"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="refresh"
    />
     Regenerate Token
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="primary"
    className="js-csv"
    disabled={false}
    download="gslt.csv"
    href="data:text/csv;charset=UTF-8,﻿%22SteamID%22%2C%22Token%22%2C%22AppID%22%2C%22Last-Logon%22%2C%22Memo%22%0A"
    title="CSV"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="download"
    />
     CSV
  </Button>,
]
`;

exports[`<List> empty items select all via button click 1`] = `
Array [
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    className="js-select-button"
    disabled={false}
    onClick={[Function]}
  >
    <Checkbox
      bsClass="checkbox"
      checked={true}
      className="js-select-checkbox"
      disabled={false}
      inline={false}
      title=""
    />
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="danger"
    className="js-remove"
    disabled={false}
    onClick={[Function]}
    title="Remove"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="trash"
    />
     Remove
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="success"
    className="js-regenerate"
    disabled={true}
    onClick={[Function]}
    title="Regenerate Token"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="refresh"
    />
     Regenerate Token
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="primary"
    className="js-csv"
    disabled={false}
    download="gslt.csv"
    href="data:text/csv;charset=UTF-8,﻿%22SteamID%22%2C%22Token%22%2C%22AppID%22%2C%22Last-Logon%22%2C%22Memo%22%0A"
    title="CSV"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="download"
    />
     CSV
  </Button>,
]
`;

exports[`<List> with a single account deselect by entry 1`] = `
Array [
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    className="js-select-button"
    disabled={false}
    onClick={[Function]}
  >
    <Checkbox
      bsClass="checkbox"
      checked={false}
      className="js-select-checkbox"
      disabled={false}
      inline={false}
      title=""
    />
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="danger"
    className="js-remove"
    disabled={false}
    onClick={[Function]}
    title="Remove"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="trash"
    />
     Remove
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="success"
    className="js-regenerate"
    disabled={false}
    onClick={[Function]}
    title="Regenerate Token"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="refresh"
    />
     Regenerate Token
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="primary"
    className="js-csv"
    disabled={false}
    download="gslt.csv"
    href="data:text/csv;charset=UTF-8,﻿%22SteamID%22%2C%22Token%22%2C%22AppID%22%2C%22Last-Logon%22%2C%22Memo%22%0A"
    title="CSV"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="download"
    />
     CSV
  </Button>,
]
`;

exports[`<List> with a single account regenerate all 1`] = `
Array [
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    className="js-select-button"
    disabled={false}
    onClick={[Function]}
  >
    <Checkbox
      bsClass="checkbox"
      checked={false}
      className="js-select-checkbox"
      disabled={false}
      inline={false}
      title=""
    />
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="danger"
    className="js-remove"
    disabled={false}
    onClick={[Function]}
    title="Remove"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="trash"
    />
     Remove
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="success"
    className="js-regenerate"
    disabled={false}
    onClick={[Function]}
    title="Regenerate Token"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="refresh"
    />
     Regenerate Token
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="primary"
    className="js-csv"
    disabled={false}
    download="gslt.csv"
    href="data:text/csv;charset=UTF-8,﻿%22SteamID%22%2C%22Token%22%2C%22AppID%22%2C%22Last-Logon%22%2C%22Memo%22%0A"
    title="CSV"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="download"
    />
     CSV
  </Button>,
]
`;

exports[`<List> with a single account remove all 1`] = `
Array [
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    className="js-select-button"
    disabled={false}
    onClick={[Function]}
  >
    <Checkbox
      bsClass="checkbox"
      checked={false}
      className="js-select-checkbox"
      disabled={false}
      inline={false}
      title=""
    />
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="danger"
    className="js-remove"
    disabled={false}
    onClick={[Function]}
    title="Remove"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="trash"
    />
     Remove
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="success"
    className="js-regenerate"
    disabled={false}
    onClick={[Function]}
    title="Regenerate Token"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="refresh"
    />
     Regenerate Token
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="primary"
    className="js-csv"
    disabled={false}
    download="gslt.csv"
    href="data:text/csv;charset=UTF-8,﻿%22SteamID%22%2C%22Token%22%2C%22AppID%22%2C%22Last-Logon%22%2C%22Memo%22%0A"
    title="CSV"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="download"
    />
     CSV
  </Button>,
]
`;

exports[`<List> with a single account select all 1`] = `
Array [
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    className="js-select-button"
    disabled={false}
    onClick={[Function]}
  >
    <Checkbox
      bsClass="checkbox"
      checked={true}
      className="js-select-checkbox"
      disabled={false}
      inline={false}
      title=""
    />
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="danger"
    className="js-remove"
    disabled={false}
    onClick={[Function]}
    title="Remove"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="trash"
    />
     Remove
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="success"
    className="js-regenerate"
    disabled={false}
    onClick={[Function]}
    title="Regenerate Token"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="refresh"
    />
     Regenerate Token
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="primary"
    className="js-csv"
    disabled={false}
    download="gslt.csv"
    href="data:text/csv;charset=UTF-8,﻿%22SteamID%22%2C%22Token%22%2C%22AppID%22%2C%22Last-Logon%22%2C%22Memo%22%0A%22212V16ECZ4HE%22%2C%227FJS3VY2273L%22%2C740%2C%221995-12-17T03%3A24%3A00.000Z%22%2C%22CSGO%22"
    title="CSV"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="download"
    />
     CSV
  </Button>,
]
`;

exports[`<List> with undefined lastLogon 1`] = `
Array [
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    className="js-select-button"
    disabled={false}
    onClick={[Function]}
  >
    <Checkbox
      bsClass="checkbox"
      checked={true}
      className="js-select-checkbox"
      disabled={false}
      inline={false}
      title=""
    />
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="danger"
    className="js-remove"
    disabled={false}
    onClick={[Function]}
    title="Remove"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="trash"
    />
     Remove
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="success"
    className="js-regenerate"
    disabled={false}
    onClick={[Function]}
    title="Regenerate Token"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="refresh"
    />
     Regenerate Token
  </Button>,
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="primary"
    className="js-csv"
    disabled={false}
    download="gslt.csv"
    href="data:text/csv;charset=UTF-8,﻿%22SteamID%22%2C%22Token%22%2C%22AppID%22%2C%22Last-Logon%22%2C%22Memo%22%0A%22212V16ECZ4HE%22%2C%227FJS3VY2273L%22%2C740%2C%2C%22CSGO%22"
    title="CSV"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="download"
    />
     CSV
  </Button>,
]
`;
